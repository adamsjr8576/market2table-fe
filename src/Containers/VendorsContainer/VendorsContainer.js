import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import './VendorsContainer.scss';
import { VendorCard } from '../../Components/VendorCard/VendorCard';
import { withRouter } from 'react-router-dom';



export const VendorsContainer = ({history}) => {
  const vendors = useSelector(state => state.vendors);
  const displayVendors = vendors.map(vendor => {
    return (
      <VendorCard
      id={vendor.id}
      name={vendor.name}
      description={vendor.description}
      image={vendor.image}
      history={history}
      />
    )
  });
    
  return (
    <div className='vendor-container'>
      {displayVendors}
    </div>
  )
}

export default withRouter(VendorsContainer);
