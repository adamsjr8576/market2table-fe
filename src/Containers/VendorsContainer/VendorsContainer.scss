.vendor-container {
  display: flex;
  flex-wrap: wrap;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  height: 80vh;
  align-items: center;
  text-align: center;
  overflow: scroll;
  margin: 0;
  padding: 0;
  justify-content: space-evenly;
  }