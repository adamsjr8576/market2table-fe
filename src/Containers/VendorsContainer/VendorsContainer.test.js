import React from 'react';
import { shallow } from 'enzyme';
import VendorsContainer from './VendorsContainer';
import { useSelector, useDispatch } from 'react-redux';

const mockState = {
    vendors: [{name:'vendorA', description:'selling', products:'apples'},{name:'vendorB', description:'sellin', products:'oranges'}],
  }

jest.mock("react-redux", () => ({
    useSelector: () => mockState,
}));

describe('VendorsContainer', () => {
    let mockHistory = {
        path:'/markets'
    };
    
    it('should match the VendorsContainer snapshot', () => {
        let wrapper = shallow(<VendorsContainer
            vendors={mockState.vendors}
            history={mockHistory}
          />);
        expect(wrapper).toMatchSnapshot();
    })

});