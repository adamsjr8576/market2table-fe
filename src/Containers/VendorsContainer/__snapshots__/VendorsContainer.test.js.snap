// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorsContainer should match the VendorsContainer snapshot 1`] = `
<main
  className="vendor-container"
>
  <VendorCard
    description="selling"
    history={
      Object {
        "path": "/markets",
      }
    }
    id="2342"
    key="2342"
    name="vendorA"
  />
  <VendorCard
    description="sellin"
    history={
      Object {
        "path": "/markets",
      }
    }
    id="123"
    key="123"
    name="vendorB"
  />
</main>
`;
